File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
            vector<int> ans; int c=0;
      ListNode* temp=head;
      while(temp!=NULL){
          ans.push_back(temp->val);
          temp=temp->next;
      }
      int i=0,j=ans.size()-1;
      while(i<j){
         if(ans[i]+ans[j]>c)
           c=ans[i]+ans[j];
           i++;
           j--;
      }
      return c;
    }
};

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        li=[]
        temp=head
        max=0
        while temp:
            li.append(temp.val)
            temp=temp.next
        i,j=0,len(li)-1
        while i<j:
            if li[i]+li[j]>max:
                max=li[i]+li[j]
            i+=1
            j-=1
        return max